hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/BulkInsertPartitioner.java

```diff
@@ -72,4 +79,37 @@ default Option<WriteHandleFactory> getWriteHandleFactory(int partitionId) {
     return Option.empty();
   }
 
+  /*
+   * If possible, we want to sort the data by partition path. Doing so will reduce the number of files written.
```

Comment:
Let's also elaborate that prepending partition columns wouldn't affect sorting order for partition table and hence it's a safe operation (for partition tables only though!)

Reply:
I'm not sure what this means. What is a partition table?

Reply:
Sorry, that was a typo, i meant "partitioned tables"